ui/dashboards/src/context/DashboardProvider/dashboard-provider-api.ts

```diff
@@ -136,12 +137,11 @@ export function useDeletePanelGroupDialog() {
  * Gets an individual panel in the store. Throws if the panel can't be found.
  */
 export function usePanel(panelGroupItemId: PanelGroupItemId) {
-  const { panelGroupId, itemIndex } = panelGroupItemId;
+  const { panelGroupId, panelGroupLayoutId } = panelGroupItemId;
 
   const panel = useDashboardStore((store) => {
-    const panelRef = store.panelGroups[panelGroupId]?.items[itemIndex]?.content;
-    if (panelRef === undefined) return;
-    const panelKey = getPanelKeyFromRef(panelRef);
+    const panelKey = store.panelGroups[panelGroupId]?.itemPanelKeys[panelGroupLayoutId];
```

Comment:
Oh this reminds me, not sure if it's still an issue, but last week I saw some weird behavior where if I gave two panels the same name, there would be a collision and subsequent changes to one would effect the other (I think I created a StatChart  and GaugeChart, then they both became GaugeCharts with duplicate names / descriptions).

Reply:
Yeah, that's been in the back of my mind as something we need to fix for panel keys. Definitely still an issue.